Windows ?

jpicus opened the pull request:
Hi,

I guess this question is not brand new: is there any plans to have this project windows eligible ?

Regrads

Commit:
Fix compilation issue

Add NIL fdw_private list param to create_foreignscan_path, whatever it
is.

Commit:
Fix another API mismatch

This needs review: I don't actually know what this is supposed to do.
Probably we have to force the push down to our purposes.

Commit:
Update build instructions

Commit:
Comment out broken log trace

This can cause a segfault. Most likely a null-check is in order.

Commit:
Ignore build artifacts

Commit:
Add flags when DEBUG is set

Commit:
Add license file

jgoizueta committed:
Fix bug causing segfaults with query columns not present in foreign table

The found variable retained the last column value as default value for
next column, so missing columns where not detected and the size of the
col_position_mask and col_size_array could be inconsistent with the
number of columns.

Commit:
Merge pull request #1 from CartoDB/license

Add license file

jgoizueta committed:
Merge pull request #2 from CartoDB/fix-segfault

Fix bug causing segfaults with query columns not present in foreign t…

jgoizueta committed:
Update README to state minimum PG version required

We will be supporting only 9.5 or newer versions of PostgreSQL

jgoizueta committed:
Fix typo

jgoizueta committed:
Fix reading NULL values

jgoizueta committed:
Fix length of copied strings

There was no provision for the trailing zero of strings.
This resulted in fixed length columns (char, varchar) losing
the last character.

jgoizueta committed:
Fix type of function

jgoizueta committed:
Fix type of NULL pointers

jgoizueta committed:
Fix array copy

jgoizueta committed:
Temporary hack to avoid memory corruption problems

Heap (palloc/pfree) memory is released between calls
of the scan iteration function. PostgreSQL memory contexts should
be used to retain Lists between calls.

As a provisional workaround the number of columns is limited
and fixed-size arrays are used for column mapping and sizes.

jgoizueta committed:
Add driver, host and port parameters

jgoizueta committed:
Use a query memory context for column mask and sizes

Replaces the temporary hack using fixed size arrays

jgoizueta committed:
Fix typo

jgoizueta committed:
Remove useless code and normalize empty options as NULL

jgoizueta committed:
Rename variable for clarity

jgoizueta committed:
Fix alignment

jgoizueta committed:
Merge pull request #3 from CartoDB/fix-columns

Fix server segfault bugs

jgoizueta committed:
Remove support for old PostgreSQL versions

Currently we're targeting PG>=9.5 only
#4

jgoizueta committed:
Merge pull request #8 from CartoDB/4-pg95

Remove support for old versions of PostgreSQL

jgoizueta committed:
Normalize whitepace to use tab indentation

This is to comply with PostgreSQL convention for C code.
Tabulators (intended to be 4-space wide) are used for indentation.
The spacing has been adjusted so that tabs are used only for
indenting, while spaces are used to align with the previous line
beyond the indentation point (so that tab width can be changed
while preserving the code alignment).

See #5

jgoizueta committed:
Rename internal function using camel_case style

jgoizueta committed:
Prune DEBUG messages

Leave only some relevant messages (mostly function entry points) and
use DEBUG1 level.

jgoizueta committed:
Merge pull request #9 from CartoDB/5-gardening

Clean up the code and adapt to PostgreSQL conventions

jgoizueta committed:
Fix setting default schema name

It used svr_database before it was set.

jgoizueta committed:
Implement IMPORT FOREIGN SCHEMA

This is WIP; currently only supports
the case of a sql_query in Options

jgoizueta committed:
Renam internal functions

jgoizueta committed:
Use structure to store options

jgoizueta committed:
Remove debugging messages

jgoizueta committed:
Add table name prefix option

This allows using a prefix for the names of all tables imported
from a foreign schema.

jgoizueta committed:
Better error checking

Handle common problems (such as unavailable ODBC source) which
were being ignored.

jgoizueta committed:
Add support for no sql_query and no sql_count cases

jgoizueta committed:
Add comment

jgoizueta committed:
Use memset to initialize structure

jgoizueta committed:
Use structure copy

jgoizueta committed:
Implement the standard IMPORT FOREIGN SCHEMA cases

jgoizueta committed:
Admit sql_query with trailing semicolon

jgoizueta committed:
Add encoding option

With this option, text can be converted from an external encoding

jgoizueta committed:
Allow username and password in server definition

jgoizueta committed:
Fix use of tables list in limit to/except clauses

jgoizueta committed:
Fix: make internal functions static

jgoizueta committed:
Fix DEBUG macrro / DEBUG1 notice level mix up

The value of the DEBUG macro (which indicates debug mode)
was being passed in some cases as a notice level
(which should be one of the DEBUG1-DEBUG5 levels)

jgoizueta committed:
Remove empty string normalization to NULL

We rely on a function is_blank_string instead to chec if
strings are empty or NULL.

jgoizueta committed:
Add support for GUID columns

jgoizueta committed:
Use bool type rather than int

For clarity of intent

jgoizueta committed:
Remove debugging messages

jgoizueta committed:
Remove debugging messages

jgoizueta committed:
Replace magic numbers by macros

Commit:
USE_PGXS=1 no longer an option

Commit:
Minimal test for build

Really minimal test, to be executed like this:
```
PGUSER=postgres make installcheck
```

jgoizueta committed:
Fix: accept user and password options in user mappings

The username and password options whould be definable
in the CREATE USER MAPPING STATEMENT as well as in
CREATE SERVER and CREATE FOREIGN TABLE (note that
the latter accepts any option).

Commit:
Merge pull request #13 from CartoDB/debian-package-dh-make

Debian package dh make

Commit:
Merge pull request #12 from CartoDB/10-import-schema

Implement IMPORT FOREIGN SCHEMA

Commit:
Change version to 0.0.1

Commit:
Set v0.0.1 in Makefile as well

We should probably improve that in the long run.

Commit:
Fix typo s/0.0.0/0.0.1/

jgoizueta committed:
Update readme

jgoizueta committed:
Add README detail

Add reference to PostgreSQL encodings.

jgoizueta committed:
Undocument the column mapping feature

It's not really tested and should be reconsidered

jgoizueta committed:
Small wording change in README

jgoizueta committed:
Corrections and new requirements section

jgoizueta committed:
Merge pull request #18 from CartoDB/4-update-readme

Update readme

jgoizueta committed:
Quote connection attributes

Fixes #15

jgoizueta committed:
Handle single quotes when quoting options

Option values are quoted in SQL statements.
If an option value contains single quotes they must
be duplicated as a way of escaping them.

Fixes #19

jgoizueta committed:
:lipstick: Fix spacing around operator

jgoizueta committed:
Fix: prevent memory leak and race conditions

Avoid using a static, mutable structure to build quoted strings.
Note that the original code leaked because the initStringInfo
function does discards used memory wthout freeing it.

jgoizueta committed:
Make sure static data is not mutable

jgoizueta committed:
Make sure static data is unmutable

jgoizueta committed:
Remove spurious line

jgoizueta committed:
Conditionally emit debug message

jgoizueta committed:
Merge pull request #20 from CartoDB/15-quote-attributes

Quote connection attributes

jgoizueta committed:
Merge pull request #21 from CartoDB/19-quotes

Handle single quotes when quoting options

jgoizueta committed:
Fix declaration

jgoizueta committed:
Define arbitrary ODBC attributes with odbc_ options

jgoizueta committed:
Remove quoting from ODBC attribute values

Closes #15
The standard (curly braces) quoting has to be handled by the user,
since some ODBC drivers (notably PostgreSQL's) do not support it.
The quoting of attribute names and values is explained in the README.

jgoizueta committed:
Fix some compilation warnings

Unused variables, function prototypes, pointer types.
Closes #14

jgoizueta committed:
Merge pull request #22 from CartoDB/11-odbc-attributes

Define arbitrary ODBC attributes with odbc_ options

jgoizueta committed:
Handle partial SQLGetData results

Deal with the case that the buffer for SQLGetData is too small,
and also with missing trailing zeros.

jgoizueta committed:
Use adequate minimum buffer size for numeric data

To avoid loss of precision

jgoizueta committed:
Limit size of varying columns and buffers

Some types (e.g. PostGIS geometry columns which are mapped to text)
may have huge sized. We now use type text for those large columns
and limit the buffer size (which is no problem since
GetData by parts was implemented)

jgoizueta committed:
Fix various binary column problems

* Non-supported column types (including bit-strings) are now omitted
* bit(1) is interpreted as boolean
* Conversion to bytea (longvarbinary) is handled properly

The buf_used variable has been added to keep track of the buffer size
in case of binary data (non null-terminated) so it will be available
for future binary conversions.

jgoizueta committed:
Define macro names for magic values

jgoizueta committed:
Refactor ifs into switch/case

Commit:
Fix indentation in switch-case block

Commit:
Add TODO note for code commented out

Commit:
Merge pull request #24 from CartoDB/23-getdata

Data buffer fixes

Commit:
Merge pull request #26 from CartoDB/23-getdata-style-fixes

23 getdata style fixes

Commit:
Update author text

Commit:
Add upgrade/downgrade files for 0.1.0

  and also update the default_version

Commit:
Merge pull request #27 from CartoDB/prepare-version-0.1.0

Prepare version 0.1.0

Commit:
Adding test capability for every connector

Commit:
Added automatically generated tests for installation test

- Added templates for the installation tests using configuration files
  for every provider
- Expected part of the tests is automatically generated too
- Added all the generated content to gitignore in order to leak info

Commit:
Added fixtures and config for mysql too

Commit:
Removed postgis part by now in travis

Commit:
README and CR corrections

Commit:
Merge pull request #29 from CartoDB/testing_connectors

Testing connectors

Commit:
Update README.md

Commit:
Added test to check out the issues for this ticket

jgoizueta committed:
Fix missing schema OPTION

The CREATE FOREIGN TABLE generated by IMPORT FOREIGN SCHEMA
needs to pass the foreign schema as an option.

jgoizueta committed:
Fix tests running command in README

jgoizueta committed:
Omit the foreign schema when a blank schema options is present

This allows to omit the schema if the driver requires it.

jgoizueta committed:
Merge pull request #31 from CartoDB/30_postgres_not_existent_schema_issue

Fix missing foreign schema

Commit:
Installing HIVE

Commit:
Added driver installation and hive tests

Commit:
Merge pull request #32 from CartoDB/add_hive_to_travis

HIVE tests and travis integration

Commit:
SQL Server tests

Commit:
Merge pull request #36 from CartoDB/sql_server_tests

SQL Server tests

Commit:
Updated tests README

Commit:
Merge pull request #37 from CartoDB/update_tests_readme

Updated tests README

Commit:
Updated freetds package version

Commit:
List tables function

Commit:
Added tests for the ODBCTablesList function

Commit:
Added functions to get the number of rows for a table or a query

Commit:
Added tests for the size functions

Commit:
Fixed problem with size pointer

Commit:
Merge pull request #38 from CartoDB/list_tables_function

List tables function

jgoizueta committed:
Fix schema returned by tables_list

Fixes #39

Commit:
Fix tests

Commit:
Merge pull request #40 from CartoDB/39-tables-list-schema

Fix schema returned by tables_list

jgoizueta committed:
Quote options to preserve case

Commit:
Merge pull request #41 from CartoDB/options_case

Quote options to preserve case

jgoizueta committed:
Add note about LIMIT TO and prefix use in IMPORT FOREIGN SCHEMA

Commit:
Add old version upgrade files

jgoizueta committed:
Fix travis problems

Commit:
Added CARTO as part of the organization in the LICENSE file

Commit:
Contributing document

Commit:
Fix travis build changing postgresql version from launchpad

Commit:
Changed to apache hive 2.1.1

The prior version was removed from the official repository

Commit:
Merge pull request #52 from CartoDB/fix_travis

Fix travis build changing postgresql version from launchpad

Commit:
Issue template added

Commit:
Contributing doc changes requested by reviewer

Commit:
Merge pull request #50 from CartoDB/contributing_document

Contributing document

bluthg commented:
TBTH, I have no idea... please kindly ask the CartoDB people, as they picked this up quite some time ago.

Commit:
Merge pull request #51 from CartoDB/license_document

Added CARTO as part of the organization in the LICENSE file

robe2 committed:
Ignore mingw64, windows, and Mac build artifacts

When I compiled under mingw64 (building for windows), it generates a libodbc_fdw.a and odbc_fdw.dll at the root of tree.  For extra measure I added .dylib since I think MacOSX generates those.  I don't have a Mac to test that.

This patch  is to ignore these build artifacts from being accidentally committed.

pramsey committed:
Remove compile errors

pramsey committed:
Fix compile warnings

pramsey committed:
Set sudo to true

pramsey committed:
Push forward to pgsql 9.6

pramsey committed:
Try forcing use of 'precise'

pramsey committed:
Packages as addons?

pramsey committed:
More addons

pramsey committed:
mysql versions explicit

pramsey committed:
Mysql 5.6

pramsey committed:
Update DL link for hive

pramsey committed:
Fix dts packaging name

pramsey committed:
Add freedts-bin

pramsey committed:
Try to ensure connections work

pramsey committed:
Try and get more LOGGING on failure of postgres init

pramsey committed:
Try to echo test

pramsey committed:
Debug flag

pramsey committed:
Add ODBC debugging flag

pramsey committed:
Strip down to just PgSQL tests

pramsey committed:
Add sqlserver packages back in

Commit:
test steps to figure out the problem

pramsey committed:
Bump up travis-ci tests to v10

pramsey committed:
Use specific odbc driver

pramsey committed:
Merge pull request #64 from robe2/patch-1

 Ignore mingw64, windows, and Mac build artifacts

pramsey committed:
Merge branch 'master' of github.com:CartoDB/odbc_fdw into pglsq10-support

pramsey committed:
Replace DEBUG blocks w/ macro function

pramsey committed:
Change TRUE/FALSE to true/false for future happiness with PgSQLv11

Commit:
Updated README.md with driver versions

Commit:
:lipstick:

Commit:
Merge pull request #67 from CartoDB/travis-fix

Travis-CI Fixes

pramsey committed:
Merge branch 'master' of github.com:CartoDB/odbc_fdw into pglsq10-support

pramsey committed:
Break out install steps into scripts for
"easier" use with a build matrix

pramsey committed:
Add keys to pgdg repo

pramsey committed:
Try to run script w/o sudo wrapper

pramsey committed:
Up version for testing next rev

pramsey committed:
And now try tests against pg10

pramsey committed:
Debug v10 failure

pramsey committed:
Change pgsql startup seq

pramsey committed:
Put all PgSQL versions into a matrix

pramsey committed:
Merge pull request #69 from CartoDB/pglsq10-support

PgSQL 10 Support

pramsey committed:
Merge branch 'master' of github.com:CartoDB/odbc_fdw into debug-macros

pramsey committed:
Enforce formatting of c/h/sql/yml files

pramsey committed:
Update editor config and apply bulk formatting pass (once only!) to get everything lined up

pramsey committed:
Merge pull request #70 from CartoDB/debug-macros

Replace DEBUG blocks w/ macro function

bluthg commented:
As discussed with @pramsey , closing this should reduce the noise...
